packages/smooth_app/lib/background/background_task_download_products.dart

```diff
@@ -0,0 +1,176 @@
+import 'dart:convert';
+
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:smooth_app/background/background_task.dart';
+import 'package:smooth_app/background/background_task_progressing.dart';
+import 'package:smooth_app/background/operation_type.dart';
+import 'package:smooth_app/database/dao_product.dart';
+import 'package:smooth_app/database/dao_work_barcode.dart';
+import 'package:smooth_app/database/local_database.dart';
+import 'package:smooth_app/query/product_query.dart';
+
+/// Background progressing task about downloading products.
+class BackgroundTaskDownloadProducts extends BackgroundTaskProgressing {
+  BackgroundTaskDownloadProducts._({
+    required super.processName,
+    required super.uniqueId,
+    required super.languageCode,
+    required super.user,
+    required super.country,
+    required super.stamp,
+    required super.work,
+    required super.pageSize,
+    required super.totalSize,
+    required this.downloadFlag,
+  });
+
+  BackgroundTaskDownloadProducts.fromJson(Map<String, dynamic> json)
+      : downloadFlag = json[_jsonTagDownloadFlag] as int,
+        super.fromJson(json);
+
+  /// Download flag. Normal case: 0, meaning all fields are downloaded.
+  final int downloadFlag;
+
+  /// Download flag mask: exclude the Knowledge Panels field from the download.
+  static const int flagMaskExcludeKP = 1;
```

Comment:
Why not a bool

Reply:
> Why not a bool

So far a `bool` would work the same, indeed.
But as it's a shared code that can be used in different situations, e.g. for a full refresh or for the top N products, I prefer to use a single `int` that potentially means dozens of `bool`s (e.g. 16+8+4+2+1) instead of adding each time a new `bool` parameter.